src/relay/ir/expr.cc

```diff
@@ -121,6 +129,39 @@ FuncType FunctionNode::func_type_annotation() const {
   return FuncTypeNode::make(param_types, this->ret_type, this->type_params, {});
 }
 
+NodeRef FunctionNode::GetAttr(const std::string& key) const {
+  if (!this->attrs.defined()) { return NodeRef(); }
+
+  const DictAttrsNode* dict_attrs = this->attrs.as<DictAttrsNode>();
+  CHECK(dict_attrs);
+  auto it = dict_attrs->dict.find(key);
+  if (it != dict_attrs->dict.end()) {
+    return (*it).second;
+  } else {
+    return NodeRef();
+  }
+}
+
+Function FunctionNode::SetAttr(const std::string& key, const NodeRef& data) const {
```

Comment:
Should this be a static function?

Reply:
I moved the helpers outside of the node in the new commits.